src/PhpSpec/CodeAnalysis/AccessInspectorInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Phpspec\CodeAnalysis;
+
+interface AccessInspectorInterface
+{
+    /**
+     * @param object $object
+     * @param string $property
+     * @param bool $withValue
+     * @return bool
+     */
+    public function isPropertyAccessible($object, $property, $withValue);
```

Comment:
Now I look at this again... this method is overloaded.

What we're interested in is whether a property is `writable` or `readable`; these should be separate methods (which would probably have some shared functionality internally).

Similarly the other method should perhaps reflect that we care it's `callable`